src/inference/src/dev/core_impl.cpp

```diff
@@ -1516,7 +1540,16 @@ ov::AnyMap ov::CoreImpl::create_compile_config(const ov::Plugin& plugin, const o
     return compile_config;
 }
 
-void ov::CoreImpl::CoreConfig::set_and_update(ov::AnyMap& config) {
+ov::CoreConfig::CoreConfig(const CoreConfig& other) {
+    {
+        std::lock_guard<std::mutex> lock(other._cacheConfigMutex);
+        _cacheConfig = other._cacheConfig;
+        _cacheConfigPerDevice = other._cacheConfigPerDevice;
+    }
+    _flag_enable_mmap = other._flag_enable_mmap;
```

Comment:
why this flag is not set within `lock` scope ?

Reply:
Lock used only for properties guarded by mutex.